atlasdb-client/src/main/java/com/palantir/atlasdb/util/SlidingWindowWeightedMeanGauge.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.util;
+
+import com.codahale.metrics.Gauge;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.google.common.collect.ImmutableList;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import org.immutables.value.Value;
+
+/**
+ * A gauge that calculates the weighted mean of updates during a sliding window of time. Correctness is only guaranteed
+ * as long as the sum of recorded weights and sum of weighted values do not exceed {@link Long#MAX_VALUE} and
+ * {@link Double#MAX_VALUE}, respectively.
+ */
+public class SlidingWindowWeightedMeanGauge implements Gauge<Double> {
+    private final Cache<Long, WeightedEntry> updates;
+    private final AtomicLong counter = new AtomicLong();
```

Comment:
Rename to something just so that we know the purpose here is literally just as arbitrary keys for the cache.